enhancements/console/dynamic-plugins.md

```diff
@@ -534,6 +602,57 @@ Both `v1` and `v1alpha1` version are supported. `v1alpha1` plugins will get
 converted by the conversion webhook server into `v1` representation.
 Conversion webhook server is part of the `console-operator` pod.
 
+#### Content Security Policy
+For Content Security Policy (CSP) feature to be considered stable, the following
+requests need to be met:
+
+##### 1. Documentation updates
+
+* Update the OpenShift official documentation to include detailed guidelines
+on configuring `ConsolePluginCSP` in the `ConsoleDynamicPlugin` CRD, along with
+recommendations.
+PR link: TBD
+
+##### 2. Release notes updates
+
+* Add CSP feature to release notes.
+  * Pull request link: TBD
+
+##### 3. Extending integration and unit test suite
+
+* Extend integration and unit test suite for console-operator repository:
+    * Standart use case, with a single plugin setting valid sources and
+    validating the final CSP result set in the `console-config.yaml` file.
+    * Edge use cases, such as many plugins defining CSPs with overlapping sources,
+    and validates the final merged CSP result, set in the `console-config.yaml` file.
+    * Pull request link: https://github.com/openshift/console-operator/pull/938
+
+##### 4. Updates to demo plugin
+
+* Update one console-demo-plugin to use the new `ConsolePluginCSP` field
+in order to demonstrate the usage.
+  * Pull request link: TBD
+
+##### 5. Integration tests updates
+
+* Extend integration test suite in console repository CI to check for CSP violations.
+* PR link: [TBD](https://issues.redhat.com/browse/CONSOLE-4279)
+
+##### 6. CI updates to all dynamic plugins
+
+* All the dynamic plugins which are enabled on the cluster by default will
+update their CI to check for CSP violations.
+* PR link: TBD
+
+Currently Console uses `Content-Security-Policy-Report-Only` instead of
+`Content-Security-Policy` header. Due to that the browser will only warn about
+Console CSP violations.
+
+To ensure adequate monitoring and adjustment, Console will continue to operate
+in `Content-Security-Policy-Report-Only` mode for at least one release after
+the above requests are met. After this period, Console will switch to using the
+`Content-Security-Policy` header, enabling full enforcement of CSP policies.
+
```

Comment:
Plugins that aren't written by RH, do they need to declare compatibility versions?

How will their authors know that they need to fix any CSP issues?

How will we make sure we don't break during upgrade when we start enforcing, will plugins have to be updated first?

Reply:
You can declare compatible OCP versions today in your plugin manifest, and the plugin will only load for those versions.

When running in development mode, plugin authors will see a clear notification when their plugin violates CSP (I believe it's a toast notification in the UI). Plugins will also include full CSP violation details in the browser console, and we're working on surfacing this in the dashboard console UI for admins (https://issues.redhat.com/browse/CONSOLE-4268). @rhamilto has started on this work.

Plugins could break when we begin enforcing CSP. Those plugins would need to be updated first. We plan to give several releases grace period for plugins to update and will monitor violations reported back to Red Hat to help us decide when to flip the switch.

Reply:
> Plugins that aren't written by RH, do they need to declare compatibility versions?
no, compatibility version should not be necessary since since the field is optional. So if a plugin wont contain the new field, it could be created. 
In case console will be in the report-only mode, the plugin will get enabled by the console, even if it will be violating the policies, in which case  the user will notified that there is a plugin violating the policies. [Here is the story](https://issues.redhat.com/browse/CONSOLE-4264)
In case of enforcing the policies, the plugin which will be violating the policies wont be enabled by the console.

> How will their authors know that they need to fix any CSP issues?
Violations will be reported in the [browser's console](https://issues.redhat.com/browse/CONSOLE-4263), [ConsolePlugin list page](https://issues.redhat.com/browse/CONSOLE-4264) and eventually even on the [Console's dashboard](https://issues.redhat.com/browse/CONSOLE-4268).

> How will we make sure we don't break during upgrade when we start enforcing, will plugins have to be updated first?
Only plugins which are enabled by default could potentially break the Console, since if they will violate the policies the wont be enabled, so Console will miss some screens. For this case we are `6. CI updates to all dynamic plugins`.